q2_mlab/doctor.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+import click
+import os
+from datetime import datetime
+
+
+def parse_info(info_filepath):
+    with open(info_filepath) as f:
+        lines = f.readlines()
+        info_dict = {
+            key.strip(): val.strip()
+            for key, val in zip(lines[0].split(","), lines[1].split(","))
+        }
+    return info_dict
+
+
+def get_uninserted_results(results_dir):
+    return [f.name for f in os.scandir(results_dir) if f.path.endswith(".qza")]
+
+
+def get_inserted_results(results_dir):
+    inserted_dir = os.path.join(results_dir, "inserted")
+    if not os.path.isdir(inserted_dir):
+        return []
+    else:
+        return [
+            f.name for f in os.scandir(inserted_dir) if f.path.endswith(".qza")
+        ]
+
+
+def filter_duplicate_parameter_results(
+    list_of_artifact_paths, results_dir, delete=False
+):
+    artifact_paths = list_of_artifact_paths.copy()
+
+    # Change working dirs
+    prev_working_dir = os.getcwd()
+    os.chdir(results_dir)
+
+    if len(artifact_paths) == 0:
+        raise ValueError("There are no result artifacts to remove.")
+    artifact_paths.sort()
+
+    curr_path = artifact_paths[0]
+    curr_param_idx = int(curr_path.split("_")[0])
+
+    for next_path in artifact_paths[1:]:
```

Comment:
Can you explain the logic in this for loop a bit more? I am not sure why you are doing timestamp compaisons

Reply:
I added a docstring that hopefully explains it.